api/v1alpha1/clusterextension_types.go

```diff
@@ -18,53 +18,35 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
-	"github.com/operator-framework/operator-controller/internal/conditionsets"
 )
 
-type UpgradeConstraintPolicy string
-
-const (
-	// The extension will only upgrade if the new version satisfies
-	// the upgrade constraints set by the package author.
-	UpgradeConstraintPolicyEnforce UpgradeConstraintPolicy = "Enforce"
-
-	// Unsafe option which allows an extension to be
-	// upgraded or downgraded to any available version of the package and
-	// ignore the upgrade path designed by package authors.
-	// This assumes that users independently verify the outcome of the changes.
-	// Use with caution as this can lead to unknown and potentially
-	// disastrous results such as data loss.
-	UpgradeConstraintPolicyIgnore UpgradeConstraintPolicy = "Ignore"
-)
+// +kubebuilder:validation:XValidation:rule="self.sourceType=='package' && has(self.__package__)",message="sourceType must match populated union field"
+//
+// ClusterExtensionSource defines the source for this ClusterExtension, right now, only a package is supported.
+type ClusterExtensionSource struct {
+	//+kubebuilder:validation:Enum:=package
+	//+kubebuilder:validation:Required
+	// SourceType is the discriminator for the source type
+	SourceType string `json:"sourceType"`
+
+	// Package defines a reference for a bundle in a catalog defined by a name and a version and/or channel
+	Package *SourcePackage `json:"package,omitempty"`
+}
```

Comment:
Heads up - this change is going have us modify the helm-poc branch.
cc: @bentito @dtfranz @tmshort